blocks/availability/availability.js

```diff
@@ -0,0 +1,181 @@
+import { createTag } from '../../scripts/scripts.js';
+
+function generateRandomNumber(max, excluding) {
+  //  minimum is inclusive, maximum is exclusive
+  let randomIndex;
+  do {
+    randomIndex = Math.floor(Math.random() * Math.floor(max));
+  } while (randomIndex === excluding);
+  return randomIndex;
+}
+
+function tooltipAfterOverride(tooltip, axis, pixels) {
```

Comment:
This method contains a lot of maths, but it is not obvious what it is doing or why it is needed...